eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala

```diff
@@ -155,8 +156,15 @@ object NodeParams {
     val maxAcceptedHtlcs = config.getInt("max-accepted-htlcs")
     require(maxAcceptedHtlcs <= Channel.MAX_ACCEPTED_HTLCS, s"max-accepted-htlcs must be lower than ${Channel.MAX_ACCEPTED_HTLCS}")
 
-    val publicAddresses = config.getStringList("server.public-ips").toList.map(ip => new InetSocketAddress(ip, config.getInt("server.port")))
-    require(publicAddresses.count(_.getAddress == null) == 0, s"server.public-ips must contain valid IP addresses or host names")
+    val publicAddresses = {
+      val errMsg = "server.public-ips must contain valid IP address(es)"
+      val publicIPs = config.getStringList("server.public-ips").toList
+      val invalid = publicIPs.filter(!InetAddresses.isInetAddress(_))
```

Comment:
I like the fact that you are not using any externale libs, but Google's Guava, which we are already using, does already exactly that:

```scala
scala> import com.google.common.net.InetAddresses
import com.google.common.net.InetAddresses
                     ^

scala> InetAddresses.forString("xx.xx.xx.xx")
java.lang.IllegalArgumentException: 'xx.xx.xx.xx' is not an IP string literal.
  at com.google.common.net.InetAddresses.formatIllegalArgumentException(InetAddresses.java:989)
  at com.google.common.net.InetAddresses.forString(InetAddresses.java:140)
  ... 64 elided

scala> InetAddresses.forString("1.2.3.4")
res2: java.net.InetAddress = /1.2.3.4
```

I'd prefer using this instead of writing our own boiler plate